ci-operator/config/openshift/csi-driver-shared-resource/openshift-csi-driver-shared-resource-master.yaml

```diff
@@ -18,6 +26,13 @@ images:
       as:
       - registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.17-openshift-4.10
   to: csi-driver-shared-resource
+- dockerfile_path: Dockerfile.mustgather
+  from: must_gather
```

Comment:
this appears to be replacing a ref to the "cli" image w/ a ref to the "must-gather" image, is that deliberate?  Should the original dockerfile be updated to ref must-gather?

https://github.com/openshift/csi-driver-shared-resource/blob/d06ff18b2ff23e5165638d6c3df42a2e1715b28f/Dockerfile.mustgather#L5

Reply:
similar monkey see monkey do

https://github.com/openshift/release/blob/master/ci-operator/config/openshift/local-storage-operator/openshift-local-storage-operator-master.yaml

and 

https://github.com/openshift/local-storage-operator/blob/master/Dockerfile.mustgather

Reply:
renaming it in the dockerfile seems logical since then:
1) you can build it locally directly out of the repository and get something closer to the same result as CI
2) clearer to people looking in the repo as to what the thing is based on

doesn't hold up this PR of course.

Reply:
it also seems strange that the dockerfile copies the entire csi-driver-shared-resource source into the image it produces, instead of just the gather script.

in fact i'm not sure i understand why that dockerfile is a multistage build at all

Reply:
yeah that question came up during the PR review for the Dockefile in the csi shared resource driver. Short answer: yes, it has to be.  I tried the various single stage permutations and none of them worked end to end.

Deeper answer: I think @adambkaplan and I had slightly different takes.  I think there are elements of how plugging into core must-gather works.  I think @adambkaplan had an idea around CI's use of golang as its base, vs. bash, wrt the entrypoint, or something like that.  He of course can clarify if I'm not capturing his thought correctly.

Reply:
i'd love to understand that better, you ought to be able to use the existing must-gather image (which gets you the oc binaries you need) as your base image, then layer on your `gather` script(that appears to be the only thing you're actually adding to the must-gather base image), then set the entrypoint to whatever you want (honestly you probably don't even need to change it, since it probably already is /usr/bin/gather for the base must-gather image)

since you're not actually doing any compiling, you don't need an intermiediary golang-based stage, that i can see.

i wonder if at least some of this is because the dockerfile itself is using the cli image instead of the must-gather image as the ostensible base (though once it's built in CI, the base ref will be subbed)

Reply:
how about this @bparees - you post the precise changes to https://github.com/openshift/csi-driver-shared-resource/blob/master/Dockerfile.mustgather you think should work here in a comment, and I'll take it, build the image, and run it against a tech preview cluster with a pod running with a shared resource volume.

I'll post the results here and we'll go from there.

Reply:
https://github.com/openshift/csi-driver-shared-resource/pull/87 is, i think, all that is needed here, but i'm trying to see if i can find an equivalent example to give me more confidence.